### PR TITLE
add package: picom

### PR DESCRIPTION
**Pull request by Yisus7u7**
It works very well, keep in mind that to use blur effects you can't use glx, you have to use xrender as a backend, even though using blur is somewhat slow.

### CODE REVIEW

**Comment by Yisus7u7** [comment]
@xeffyr I need utash to compile this package, utash are just a header for includes

https://github.com/troydhanson/uthash

File: packages/picom/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/yshui/picom
+TERMUX_PKG_DESCRIPTION="A lightweight compositor for X11"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@Yisus7u7 <jesuspixel5@gmail.com>"
+TERMUX_PKG_VERSION=8.2
+TERMUX_PKG_SRCURL=https://github.com/yshui/picom/archive/refs/tags/vNext.tar.gz
```
Comment (xeffyr) [inline packages/picom/build.sh:6]: Please never use *unversioned* tarballs.

**Review by xeffyr — CHANGES_REQUESTED** [review]

**Comment by xeffyr** [comment]
> I need utash to compile this package, utash are just a header for includes

This needs to be submitted to main repo.

**Comment by Yisus7u7** [comment]
> > Necesito utash para compilar este paquete, utash son solo un encabezado para incluye
> 
> Esto debe enviarse al repositorio principal.

Ok, I'll wait for it to be added

File: packages/picom/build.sh (hunk 1 above)
Comment (Yisus7u7) [reply packages/picom/build.sh:6]: The latest version doesn't have version ID, so I can't do it

Comment (xeffyr) [reply packages/picom/build.sh:6]: Latest is here: https://github.com/yshui/picom/archive/refs/tags/v8.2.tar.gz. The one you picked is a pre-release. Don't use nightlies, pre-releases and so on without good reason.

You can use commits to download tarball without specifying version: https://github.com/yshui/picom/archive/dac85eac10082dfc3df463aaa74b811144e22122.zip.

Source URLs should point to content that doesn't change over time.

Comment (Yisus7u7) [reply packages/picom/build.sh]: Solved

**Comment by layaachmad** [comment]
try to add more missing dependency

**Comment by leapofazzam123** [comment]
@Yisus7u7 how did your quotes become spanish

**Comment by Yisus7u7** [comment]
> @Yisus7u7 how did your quotes become spanish

I have no idea lol

**Comment by xeffyr** [comment]
`uthash` has been added 10 days ago, so what reason for waiting?

**Comment by Yisus7u7** [comment]
> `uthash` has been added 10 days ago, so what reason for waiting?

I didn't know that

File: packages/picom/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@Yisus7u7 <jesuspixel5@gmail.com>"
+TERMUX_PKG_VERSION=8.2
+TERMUX_PKG_SRCURL=https://github.com/yshui/picom/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=9d0c2533985e9670ff175e717a42b5bf1a2a00ccde5cac1e1009f5d6ee7912ec
+TERMUX_PKG_DEPENDS="libx11, libxcomposite, libxdamage, libxext, libxfixes, libxrender, xorgproto, xcb-proto, libconfig, libxdamage, libxcb, libxfixes, xcb-util-renderutil, libxrandr, libxcomposite, xcb-util-image, libxinerama, libpixman, dbus, pcre, libev, uthash"
+
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS(){
+	-Dopengl=false
+	}
```
Comment (xeffyr) [inline packages/picom/build.sh:12]: I'm strongly recommending to visit https://github.com/termux/termux-packages/wiki/Creating-new-package#table-of-available-package-control-fields and also learn shell scripting. The lines 10,11,12 contain statement that is... simply wrong.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/picom/build.sh (hunk 2 above)
Comment (Yisus7u7) [reply packages/picom/build.sh:12]: I have already checked and still do not understand the error. 
Isn't that function supposed to pass arguments to the build system?

Comment (xeffyr) [reply packages/picom/build.sh:12]: Do you understand the difference between variable and function?

Comment (xeffyr) [reply packages/picom/build.sh:12]: Even if it is a function, the command `-Dopengl=false` doesn't exist anyway. So what you have made is wrong.

However that field really is not supposed to be a function and can't be used as such. This is variable.

**Review by xeffyr — COMMENTED** [review]
Fixed `TERMUX_PKG_EXTRA_CONFIGURE_ARGS`, but really submitter must take care about such things.

**Comment by Yisus7u7** [comment]
> Fixed `TERMUX_PKG_EXTRA_CONFIGURE_ARGS`, but really submitter must take care about such things.

MThank you very much, I already understood 😅

Every day you learn something new.

**Comment by xeffyr** [comment]
Closing as inactive.

**Comment by Yisus7u7** [comment]
> Closing as inactive.

I think I already know what the problem is.

**Comment by Yisus7u7** [comment]
@xeffyr Try without extra arguments to see if it works

**Comment by xeffyr** [comment]
@Yisus7u7 I really don't know what to say... That's your PR and you should take care about it. I'm telling you this not the first time.

I guess if I wouldn't closed this pull request, you would be waited for me for months that I would come and fix everything.

pr-guideline-essentials

P.S. Error is inside `meson.build` file and not due to passed options.

**Pull request closed without merging**